src/neomaril_codex/model.py

```diff
@@ -322,42 +340,97 @@ def delete(self):
         Returns
         -------
         str
-            If model is at status=Deployed deletes the model and return a json with his information. If it isn't Deployed it returns the message that the model is under another state
+            If model is at status=Deployed deletes the model and return a json with his information.
+            If it isn't Deployed it returns the message that the model is under another state
 
         Example
         -------
         >>> model.delete()
         """
-        if self.__model_ready:
-            token = refresh_token(*self.credentials, self.base_url)
-            req = requests.delete(
-                f"{self.base_url}/model/delete/{self.group}/{self.model_id}",
-                headers={"Authorization": "Bearer " + token},
-            )
+        if self.__get_status() not in [ModelState.Disabled, ModelState.DisabledFailed, ModelState.Failed]:
```

Comment:
again, i don't think the codex should be doing these validations. its responsibility is to make requests to the API and handle the output. here we're expressing a business logic on the codex.